src/Microsoft.ML.Data/Transforms/CopyColumnsTransform.cs

```diff
@@ -163,4 +169,189 @@ protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, ou
+            {
+                if (originDic.ContainsKey(pair.source))
+                {
+                    var col = originDic[pair.source].CloneWithNewName(pair.name);
+                    resultDic[pair.name] = col;
+                }
+                else
+                {
+                    throw Contracts.ExceptUserArg(nameof(inputSchema), $"{pair.source} not found in {nameof(inputSchema)}");
+                }
+            }
+            var result = new SchemaShape(originDic.Values.ToArray());
+            return result;
+        }
+    }
+
+    public sealed class CopyColumnsTransformer : ITransformer, ICanSaveModel
+    {
+        private readonly (string Source, string Name)[] _columns;
+        private readonly IHost _host;
+
+        private class CopyColumnsRowMapper : IRowMapper
+        {
+            private readonly ISchema _schema;
+            private readonly HashSet<int> _originalColumnSources;
+            private (string source, string name)[] _columns;
+
+            public CopyColumnsRowMapper(ISchema schema, (string source, string name)[] columns)
+            {
+                _schema = schema;
+                _columns = columns;
+                _originalColumnSources = new HashSet<int>();
+                var sources = new HashSet<string>();
+                foreach (var source in columns.Select(x => x.source))
+                    sources.Add(source);
+                for (int i = 0; i < _schema.ColumnCount; i++)
+                    if (sources.Contains(_schema.GetColumnName(i)))
+                        _originalColumnSources.Add(i);
+
+            }
+
+            public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                var result = new Delegate[_columns.Length];
+                int i = 0;
+                foreach (var column in _columns)
+                {
+                    // validate activeOutput.
+                    input.Schema.TryGetColumnIndex(column.source, out int colIndex);
+                    var type = input.Schema.GetColumnType(colIndex);
+
+                    result[i++] = Utils.MarshalInvoke(MakeGetter<int>, type.RawType, input, colIndex);
+                }
+                disposer = null;
+                return result;
+            }
+
+            private Delegate MakeGetter<T>(IRow row, int src) => row.GetGetter<T>(src);
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput) => (col) => (activeOutput(col) && _originalColumnSources.Contains(col));
+
+            public RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var result = new RowMapperColumnInfo[_columns.Length];
+                // Do I need return all columns or only output???
+                for (int i = 0; i < _columns.Length; i++)
+                {
+                    _schema.TryGetColumnIndex(_columns[i].source, out int colIndex);
+                    // how to get metadata?
```

Comment:
let's add a REVIEW for now, that we want metadata to become IRow.
For now, pre-populate this output in ctor by pulling metadata from src schema